src/CommandHandler/RefundUnitsHandler.php

```diff
@@ -49,6 +67,12 @@ public function __invoke(RefundUnits $command): void
         $refundedTotal += $this->orderUnitsRefunder->refundFromOrder($command->unitIds(), $orderNumber);
         $refundedTotal += $this->orderShipmentsRefunder->refundFromOrder($command->shipmentIds(), $orderNumber);
 
+        $order = $this->orderRepository->findOneByNumber($orderNumber);
+
+        if ($this->orderFullyRefundedTotalChecker->check($order)) {
+            $this->orderFullyRefundedStateResolver->resolve($order);
```

Comment:
I think that this logic should be in separate service, that would react to `UnitsRefunded` event